ui/src/tool-panel/operational-point/operational-point-switches-infobox.tsx

```diff
@@ -0,0 +1,430 @@
+                    />
+                ))}
+            {switches.length > maxSwitchesToDisplay && (
+                <ShowMoreButton
+                    expanded={showAll}
+                    onShowMore={() => setShowAll((v) => !v)}
+                    showMoreText={t('tool-panel.operational-point.switch-links.show-all', {
+                        count: switches.length,
+                    })}
+                />
+            )}
+            {switches.length > 0 && (
+                <Button
+                    variant={ButtonVariant.GHOST}
+                    size={ButtonSize.SMALL}
+                    onClick={() => linkingAction(switches.map(({ switchItem }) => switchItem))}>
+                    {t(
+                        `tool-panel.operational-point.switch-links.${linkingDirection === 'linking' ? 'attach' : 'detach'}-all`,
+                        { count: switches.length },
+                    )}
+                </Button>
+            )}
+        </>
+    );
+};
+
+type OperationalPointSwitchRowProps = {
+    layoutContext: LayoutContext;
+    switchItem: LayoutSwitch;
+    issues: SwitchRowValidationIssue[];
+    linkingAction: (switches: { id: LayoutSwitchId; name: string }[]) => void;
+    linkingDirection: LinkingDirection;
+};
+
+const OperationalPointSwitchRow: React.FC<OperationalPointSwitchRowProps> = ({
+    layoutContext,
+    switchItem,
+    issues,
+    linkingAction,
+    linkingDirection,
+}) => {
+    return (
+        <div>
+            <SwitchBadge switchItem={switchItem} />{' '}
+            <Button
+                variant={ButtonVariant.GHOST}
+                size={ButtonSize.SMALL}
+                icon={linkingDirection === 'linking' ? Icons.Add : Icons.Subtract}
+                onClick={() => linkingAction([switchItem])}
+            />
+            {issues.slice(0, 1).map((issue, i) => (
+                <SwitchRowValidationIssueBadge
+                    key={i}
+                    issue={issue}
+                    layoutContext={layoutContext}
+                />
+            ))}
+        </div>
+    );
+};
+
+const SwitchRowValidationIssueBadge: React.FC<{
+    layoutContext: LayoutContext;
+    issue: SwitchRowValidationIssue;
+}> = ({ layoutContext, issue }) => {
+    const { t } = useTranslation();
+    const operationalPoint = useOperationalPoint(
+        'operationalPointId' in issue ? issue.operationalPointId : undefined,
+        layoutContext,
+    );
+    return (
+        <>
+            <Icons.StatusError />
+            {t(`tool-panel.operational-point.switch-links.${issue.type}`, {
+                operationalPointName: operationalPoint?.name,
+            })}
+        </>
+    );
+};
+
+type SwitchRowValidationIssue =
+    | { type: 'switch-linked-to-wrong-point'; operationalPointId: OperationalPointId }
+    | {
+          type: 'switch-in-wrong-polygon';
+          operationalPointId: OperationalPointId;
+      }
+    | { type: 'switch-not-in-polygon' };
```

Comment:
Tämä on toki oikeampi tyypitys, mutta helppokäyttöisyyden nimissä saattaisin expandata tätä määritelmää siten, että myös tuo `switch-not-in-polygon`-issuetyyppi sisältäisi `operationalPointId: undefined;`. Siten rivillä 383 ei tarvitsisi tarkastella löytyykö kenttää tyypistä, vaan voisi hakea id:n tyyliin `issue.operationalPointId`